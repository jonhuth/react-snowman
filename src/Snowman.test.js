import React from "react";
import Snowman from "./Snowman";
import { render, fireEvent, getByRole, queryByRole } from "@testing-library/react";

it("renders Snowman without crashing", function () {
  render(<Snowman />);
});

it("snapshot test of DOM at start", function () {
  const { asFragment } = render(<Snowman words={["apple"]}/>);
  expect(asFragment()).toMatchSnapshot();
})

it("snowman image changes to img1 on incorrect guess", function() {
  const { getByText, getByRole } = render(<Snowman words={["apple"]}/>);
  // incorrect ltr chosen
  const button = getByText("v");

  fireEvent.click(button);

  const img = getByRole('img');
  expect(img).toHaveAttribute("src", "1.png");
})

it("snowman image should stay as img0 on correct guess", function() {
  const { getByText, getByRole } = render(<Snowman words={["apple"]}/>);
  // incorrect ltr chosen
  const button = getByText("a");

  fireEvent.click(button);

  const img = getByRole('img');
  expect(img).toHaveAttribute("src", "0.png");
})

it("ensures guesses are limited by max number of guesses", function() {
  const { getByText, asFragment } = render(<Snowman words={["apple"]} maxWrong={6}/>);
  // incorrect ltr chosen
  const button1 = getByText("b");
  const button2 = getByText("c");
  const button3 = getByText("d");
  const button4 = getByText("f");
  const button5 = getByText("g");
  const button6 = getByText("h");

  fireEvent.click(button1);
  fireEvent.click(button2);
  fireEvent.click(button3);
  fireEvent.click(button4);
  fireEvent.click(button5);
  fireEvent.click(button6);

  expect(asFragment()).toMatchSnapshot();

  expect(getByText("Ran out of guesses")).toBeInTheDocument();
})

it("ensures game resets on reset button click", function() {
  const { getByText, queryByText } = render(<Snowman words={["apple"]} maxWrong={6}/>);
  // incorrect ltr chosen
  const ltrButton = getByText("b");
  const button1 = getByText("RESET GAME");


  fireEvent.click(ltrButton);
  expect(getByText("Number wrong: 1")).toBeInTheDocument();
  expect(queryByText("Number wrong: 0")).not.toBeInTheDocument(); 
  fireEvent.click(button1);
  expect(getByText("Number wrong: 0")).toBeInTheDocument();
  expect(queryByText("Number wrong: 1")).not.toBeInTheDocument();

})

