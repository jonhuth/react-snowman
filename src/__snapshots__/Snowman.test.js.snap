// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensures game resets on reset button click 1`] = `
<DocumentFragment>
  <div
    class="Snowman"
  >
    <img
      src="0.png"
    />
    <p>
      Number wrong: 0
    </p>
    <p
      class="Snowman-word"
    >
      _____
    </p>
    <p>
      <button
        value="a"
      >
        a
      </button>
      <button
        value="b"
      >
        b
      </button>
      <button
        value="c"
      >
        c
      </button>
      <button
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        value="f"
      >
        f
      </button>
      <button
        value="g"
      >
        g
      </button>
      <button
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <button>
      RESET GAME
    </button>
  </div>
</DocumentFragment>
`;

exports[`ensures guesses are limited by max number of guesses 1`] = `
<DocumentFragment>
  <div
    class="Snowman"
  >
    <img
      src="6.png"
    />
    <p>
      Number wrong: 6
    </p>
    <p
      class="Snowman-word"
    >
      _____
    </p>
    <p>
      Ran out of guesses
    </p>
    <button>
      RESET GAME
    </button>
  </div>
</DocumentFragment>
`;

exports[`snapshot test of DOM at start 1`] = `
<DocumentFragment>
  <div
    class="Snowman"
  >
    <img
      src="0.png"
    />
    <p>
      Number wrong: 0
    </p>
    <p
      class="Snowman-word"
    >
      _____
    </p>
    <p>
      <button
        value="a"
      >
        a
      </button>
      <button
        value="b"
      >
        b
      </button>
      <button
        value="c"
      >
        c
      </button>
      <button
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        value="f"
      >
        f
      </button>
      <button
        value="g"
      >
        g
      </button>
      <button
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <button>
      RESET GAME
    </button>
  </div>
</DocumentFragment>
`;
